### PR TITLE
Straightforward implementation of IDNA mapping, for tests only

### PR DESCRIPTION
**Pull request by swankjesse**
As described in UTS #46, https://www.unicode.org/reports/tr46

This is working towards OkHttp's own implementation of what IDN.toASCII() does on the JVM.

### CODE REVIEW

File: okhttp/src/jvmTest/java/okhttp3/internal/idn/IdnaMappingTable.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+package okhttp3.internal.idn
+
+import okio.BufferedSink
+
+interface IdnaMappingTable {
+
+  /**
+   * Returns true if the [codePoint] was applied successfully. Returns false if it was disallowed.
+   */
+  fun apply(codePoint: Int, sink: BufferedSink): Boolean
```
Comment (yschimke) [inline okhttp/src/jvmTest/java/okhttp3/internal/idn/IdnaMappingTable.kt:25]: nit: `apply` not as an operator seems a bit `doStuff`

File: okhttp/src/jvmTest/java/okhttp3/internal/idn/PlainTextIdnaMappingTable.kt [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  override fun apply(codePoint: Int, sink: BufferedSink): Boolean {
+    val index = mappings.binarySearch {
+      when {
+        it.sourceCodePoint1 < codePoint -> -1
+        it.sourceCodePoint0 > codePoint -> 1
+        else -> 0
+      }
+    }
+
+    val mapping = mappings[index]
```
Comment (yschimke) [inline okhttp/src/jvmTest/java/okhttp3/internal/idn/PlainTextIdnaMappingTable.kt:46]: Should you handle missing values before an IOOBE?

File: okhttp/src/jvmTest/java/okhttp3/internal/idn/PlainTextIdnaMappingTable.kt [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+      }
+      TYPE_DISALLOWED -> result = false
+    }
+
+    return result
+  }
+}
+
+
+private val optionsDelimeter = Options.of(
```
Comment (yschimke) [inline okhttp/src/jvmTest/java/okhttp3/internal/idn/PlainTextIdnaMappingTable.kt:65]:
```suggestion
private val optionsDelimiter = Options.of(
```

File: okhttp/src/jvmTest/java/okhttp3/internal/idn/IdnaMappingTableTest.kt [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      readPlainTextIdnaMappingTable()
+    }
+  }
+
+  @Test fun regularMappings() {
+    assertThat("hello".map()).isEqualTo("hello")
+    assertThat("hello-world".map()).isEqualTo("hello-world")
+    assertThat("HELLO".map()).isEqualTo("hello")
+    assertThat("Hello".map()).isEqualTo("hello")
+    assertThat("¼".map()).isEqualTo("1⁄4")
```
Comment (yschimke) [inline okhttp/src/jvmTest/java/okhttp3/internal/idn/IdnaMappingTableTest.kt:42]: Maybe worth a few comments. Is the former a unicode character, and the latter a combined unicode sequence?

**Review by yschimke — APPROVED** [review]

File: okhttp/src/jvmTest/java/okhttp3/internal/idn/IdnaMappingTable.kt (hunk 1 above)
Comment (swankjesse) [reply okhttp/src/jvmTest/java/okhttp3/internal/idn/IdnaMappingTable.kt:25]: Good call. Renamed to `map`

File: okhttp/src/jvmTest/java/okhttp3/internal/idn/PlainTextIdnaMappingTable.kt (hunk 3 above)
Comment (swankjesse) [reply okhttp/src/jvmTest/java/okhttp3/internal/idn/PlainTextIdnaMappingTable.kt:65]: Nice catch. Fixed

File: okhttp/src/jvmTest/java/okhttp3/internal/idn/PlainTextIdnaMappingTable.kt (hunk 2 above)
Comment (swankjesse) [reply okhttp/src/jvmTest/java/okhttp3/internal/idn/PlainTextIdnaMappingTable.kt:49]: YES! Done.

File: okhttp/src/jvmTest/java/okhttp3/internal/idn/IdnaMappingTableTest.kt (hunk 4 above)
Comment (swankjesse) [reply okhttp/src/jvmTest/java/okhttp3/internal/idn/IdnaMappingTableTest.kt:45]: Added!

**Pull request merged**